tesseract_core/sdk/tesseract.py

```diff
@@ -62,49 +63,77 @@ class Tesseract:
 
     def __init__(self, url: str) -> None:
         self._spawn_config = None
-        self._project_id = None
-        self._container_id = None
+        self._serve_context = None
         self._lastlog = None
         self._client = HTTPClient(url)
 
+    @classmethod
+    def from_url(cls, url: str) -> Tesseract:
+        obj = cls.__new__(cls)
+        obj.__init__(url)
+        return obj
+
     @classmethod
     def from_image(
         cls,
         image: str,
         *,
         volumes: list[str] | None = None,
         gpus: list[str] | None = None,
-    ):
+    ) -> Tesseract:
         obj = cls.__new__(cls)
         obj._spawn_config = SpawnConfig(
             image=image,
             volumes=volumes,
             gpus=gpus,
             debug=True,
         )
-        obj._project_id = None
-        obj._container_id = None
+        obj._serve_context = None
         obj._lastlog = None
         obj._client = None
         return obj
 
+    @classmethod
+    def from_tesseract_api(
+        cls,
+        tesseract_api_path: str | Path,
+    ) -> Tesseract:
+        """Create a Tesseract instance from a tesseract API path.
+
+        Warning: This does not use a containerized Tesseract, but rather
+        imports the Tesseract API directly. This is useful for debugging,
+        but requires a matching runtime environment + all dependencies to be
+        installed locally.
+
+        Args:
+            tesseract_api_path: Path to the tesseract API.
+
+        Returns:
+            A Tesseract instance.
+        """
+        obj = cls.__new__(cls)
+        obj._spawn_config = None
+        obj._serve_context = None
+        obj._lastlog = None
+        obj._client = LocalClient(tesseract_api_path)
+        return obj
+
     def __enter__(self):
-        if self._client is not None:
+        if self._serve_context is not None:
             raise RuntimeError("Cannot nest `with Tesseract ...` context managers. ")
```

Comment:
I just noticed that this error is misleading (one could nest two different Tesseract contexts, one for tesseract one and the other for tesseract 2); I will create an issue